### DISCUSSION TITLE
Вопрос по fs модулю или его аналогам

### DISCUSSION THREAD

**Discussion by ImLewel** [HowProgrammingWorks/LiveQA#37] [Q&A]
Можно ли с помощью ноды на пк достучаться до файловой системы андроида через модуль fs? Если нет, то есть ли его аналоги, которые так могут?

**Comment by tshemsedinov** [comment]
Нет конечно. Но вопрос неоднозначный, с одной стороны android это тоже linux и есть возможность запустить node.js на android, я такое делал просто поиграться, и там у fs есть доступ к файловой системе. Но что-то мне подсказывает, что вы не это спрашиваете, а хотите, чтобы человек из браузера на андроиде зашел к вам на страницу, которая отдается нодой с сервера, и вы с сервера дотянулись к нему в локальную файловую систему. Это, естественно, невозможно и даже вопрос так стоять не может. Вы представьте, что было бы с безопасностью, если бы так можно было делать. Но в браузере есть Filesystem WEB API - https://developer.mozilla.org/en-US/docs/Web/API/FileSystem и вообще, сайты могут запросить пользователя работать или с песочницей или со всех файловой системой, но для этого нужно дать сайту и браузеру, кстати, специальные права.

**Reply by ImLewel** [reply]
Нет, я не хочу чтобы с браузера кто то заходил. Есть консольное приложение, прописываю путь к файлу на андроиде и нужно его прочитать и перезаписать. Думал, что дело в разности путей виндовса и андроида(линукс), но проверил с линукса и ничего не поменялось. В интернете маловато информации по этому поводу нашел, может не так искал но запрос был подобен тому что здесь.
P.S. где то прочел что у ноды на андроид есть доступ только в пределах папки приложения

**Reply by tshemsedinov** [reply]
Теперь вопрос еще более непонятен. Консольное приложение где запускается? Где андроид? Куда прописывается путь?

**Reply by ImLewel** [reply]
Консольное приложение запускается с пк, путь к файлам на андроиде прописывается так же в этом приложении

**Reply by tshemsedinov** [reply]
А как пк и андроид связаны?

**Reply by ImLewel** [reply]
По usb телефон подключен к пк

**Reply by ImLewel** [reply]
Я же могу получить доступ к флэшке через ноду, андроид к пк подключен подобно флэшке по сути

**Reply by Dnkfff** [reply]
Не знаю как на андроидах, но как по мне лучше использовать какой-то фреймворк, что б напрямую нормально общаться с телефоном. Для андроида есть очень много удобных приколях, но если ты копаешь в сторону JS и ноды, советую попробовать **React Native**. Если всё таки нет желания углубляться в столь сырой и свежий продукт, советую изучить **Java** или **Kotlin**

Уверен, есть много аналогов, но зачем пытаться использовать фс для телефона? xD

**Reply by ImLewel** [reply]
Хотел использовать фс или его аналог, какая разница почему я сделал такой выбор. Вопрос был о существовании способов именно для ноды

**Reply by tshemsedinov** [reply]
Если по проводу, то телефон монтируется к компу как флешка, для fs нет разницы, можно читать как обычную файловую систему

**Reply by ImLewel** [reply]
Мне ошибку кидает, может надо как-то иначе путь указывать?

**Reply by ImLewel** [reply]
https://askubuntu.com/questions/596441/how-to-access-my-android-phone-from-my-terminal
Как оказалось, получить доступ к устройству куда сложнее. Что на виндовс что на линукс нода автоматически не получит путь, а сам телефон не назначается какой-либо буквой как диск. В этом посте сказано что нужно использовать mtp, это усложняет задачу как минимум тем, что я не знаю как из ноды прописать комманду в обычный терминал и оттуда получить данные. Скорее всего это и есть причина ошибки и частично ответ на мой вопрос. В любом случае, спасибо за помощь

**Comment by tshemsedinov** [comment]
Где код? Где ошибка? Присылай, на практические вопросы без деталей ответить нельзя

**Reply by ImLewel** [reply]
https://github.com/ImLewel/SSHack код здесь.
Ошибка пишет что не может найти путь, позже скину. Если делать все то же самое, только с файлами на пк, то все нормально.
